opm/simulators/utils/DamarisVar.hpp

```diff
@@ -0,0 +1,620 @@
+                        resbool = false ;
+                        has_error_ = true ;
+                    }
+                }
+                if (resbool == true) 
+                    paramaters_set_ = true ;
+            }
+
+            /**
+            *  Method to set the Damaris position values.
+            *
+            *  /param [IN] positionsVals : An pointer to a value or array of values to set as the offset into the array. 
+            *                              One element per dimension (one value for each dim_)
+            *
+            *  /implicit                 : Implicitly uses the variable name: \ref variable_name_
+            */
+            void SetDamarisPosition( std::vector<int64_t> positionsVals ) 
+            {
+                assert(positionsVals.size() == dims_) ;
+
+                for (int pos_dim = 0 ; pos_dim < dims_ ; pos_dim++)
+                {
+                    positions_[pos_dim] = positionsVals[pos_dim] ;
+                }
+                dam_err_ = damaris_set_position(variable_name_.c_str(), positionsVals.data());
+                if (dam_err_ != DAMARIS_OK) {
+                    dam_err_sstr_ << "  ERROR rank =" << rank_ << " : class DamarisVar : damaris_set_position(\"" 
+                                     << variable_name_ << "\", positionsVals);  Damaris error = " <<  damaris_error_string(dam_err_) << std::endl ;
+                    has_error_ = true ;
+                }
+            }
+
+            /**
+            *  Method to set the Damaris paramater values.
+            *
+            *  /implicit                : Implicitly uses the Damaris variable name string  \ref variable_name_
+            *  /implicit                : Implicitly uses the class data element : \ref data_ptr_
+            */
+            void SetPointersToDamarisShmem( void )
+            {
+                if (paramaters_set_ == true )
+                {
+                    // Allocate memory in the shared memory section... 
+                    dam_err_ = damaris_alloc(variable_name_.c_str(), (void **) &data_ptr_) ;
+                    if (dam_err_ != DAMARIS_OK) {
+                        dam_err_sstr_ << "  ERROR rank =" << rank_ << " : class DamarisVar : damaris_alloc(\"" 
+                                      << variable_name_ <<"\", (void **) &ret_ptr)" << ", Damaris error = " <<  damaris_error_string(dam_err_) << std::endl ;
+                        has_error_ = true ;
+                    }
+                } else {
+                    dam_err_ = -1 ;
+                    dam_err_sstr_ << "ERROR rank =" << rank_  << " : class DamarisVar : SetDamarisParameter() should be called first so as to define the size of the memory block required for variable : " << variable_name_ << std::endl ;
+                    has_error_ = true ;
+                }
+            }
+
+            void SetPointersToDamarisShmem( T ** ptr_in )
```

Comment:
What is the use case for this overload? Can it be removed? I think an outside caller could just get the data pointer using the accessor if necessary.